### ISSUE TITLE
Peak memory usage with DataBackendSession.to_query_result()

### ISSUE THREAD

**Issue by ghill2** [nautechsystems/nautilus_trader#1253]
# Bug Report

### Expected Behavior
The DataBackendSession.to_query_result() should stream chunks from the file with peak memory usage staying low and consistent

### Actual Behavior
The DataBackendSession.to_query_result() shows high peak memory usage at the first chunk generated from the method.
Peak memory measured at the first chunk: 2GB.
Parquet file size: 720MB.
Additionally, the peak memory usage was also confirmed by monitoring the process in macOS Activity Monitor application.

### Steps to Reproduce the Problem

parquet file link: https://drive.google.com/file/d/1SSiXEgronkbiBbhJ05H4Im-ux0fPOH6x/view?usp=sharing

```python
import pandas as pd
from nautilus_trader.core.datetime import dt_to_unix_nanos
from nautilus_trader.core.nautilus_pyo3.persistence import DataBackendSession
from nautilus_trader.core.nautilus_pyo3.persistence import NautilusDataType
import psutil
import platform
from pathlib import Path

def get_peak_memory_usage_gb():
    BYTES_IN_GIGABYTE = 1e9
    if platform.system() == "Darwin" or platform.system() == "Linux":
        import resource
        return resource.getrusage(resource.RUSAGE_SELF).ru_maxrss / BYTES_IN_GIGABYTE
    elif platform.system() == "Windows":
        return psutil.Process().memory_info().peak_wset / BYTES_IN_GIGABYTE
    else:
        raise RuntimeError("Unsupported OS.")
    
def test_streaming_memory_usage(*args, **kwargs):
    path = str(Path("~/Desktop/GBPUSD-DUKA-FX-QUOTETICK-1-TICK-ASK-2022.parquet").expanduser())
    session = DataBackendSession(chunk_size=5_000)
    table = "quotes"
    start_ts = dt_to_unix_nanos(pd.Timestamp("2022-03-01"))
    end_ts = dt_to_unix_nanos(pd.Timestamp("2022-07-01"))
    query = f"SELECT * FROM {table} WHERE ts_init >= {start_ts} AND ts_init <= {end_ts} ORDER BY ts_init"
    session.add_file_with_query(table, path, query, NautilusDataType.QuoteTick)
    for chunk in session.to_query_result():
        print(get_peak_memory_usage_gb())
        exit()
        

if __name__ == "__main__":
    test_streaming_memory_usage()
    
```

### Specifications

- OS platform:
- macOS Monterey 12.2
- Python version:
- 3.10
- `nautilus_trader` version:
- 1.179.0 develop branch

**Comment by cjdsellers** [comment]
Hi @ghill2 

Thanks for the report with MRE, we'll investigate this!

**Comment by cjdsellers** [comment]
Now closed from 783740e40f10dac79cbf4bb0b7d0012a7bd4276b, many thanks @twitu!
